### PR TITLE
Add external libraries article

### LINKED ISSUE #167: Document how to include C++ and Java Libraries in robot code
I received this email message.  It seems like something we should include in the documentation.

>Good afternoon,
>I have a question regarding how to include compiled libraries with gradle so they compile during the build process. I am trying to include a .a file while building code in vs code (for a c++ frc project), and we cannot find the correct documentation as to what command to use and where it needs to be. Could you point to where this documentation is for the c++ implementation of gradle?
>Thank you in advance,

### PR DESCRIPTION
**Pull request by Daltz333**
Close #167 

Preview: https://frc-docs--1218.org.readthedocs.build/en/1218/docs/software/advanced-gradlerio/index.html

### CODE REVIEW

File: source/docs/software/advanced-gradlerio/external-libraries.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Copying Source Code
+^^^^^^^^^^^^^^^^^^^
+
+Simply copy the necessary source and/or headers into your robot project. You can then configure any necessary platform args like below:
+
+.. code-block:: groovy
+
+   nativeUtils.platformConfigs.named("linuxathena").configure {
+      it.cppCompiler.args.remove('-g')
+   }
```
Comment (ThadHouse) [inline source/docs/software/advanced-gradlerio/external-libraries.rst:52]: This is completely unrelated to copying in source code. Removing -g just removes debug info.

Comment (ThadHouse) [reply source/docs/software/advanced-gradlerio/external-libraries.rst:52]: You didn't remove it. Its still there for athena...

File: source/docs/software/advanced-gradlerio/external-libraries.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
+Using External Libraries with Robot Code
+========================================
+
+.. warning:: Using external libraries may have unintended behavior with your robot code! It is not recommended unless you are aware of what you are doing!
+
+Often a team might want to add external Java or C++ libraries for usage with their robot code. This article highlights adding Java libraries to your gradle dependencies, or the options that C++ teams have.
```
Comment (Starlight220) [inline source/docs/software/advanced-gradlerio/external-libraries.rst:6]:
```suggestion
Often a team might want to add external Java or C++ libraries for usage with their robot code. This article highlights adding Java libraries to your Gradle dependencies, or the options that C++ teams have.
```
Capitalize Gradle? If we standardize on uppercase Gradle, then `gradle` should be added to the spellchecker.

File: source/docs/software/advanced-gradlerio/external-libraries.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+Robot projects by default do not have a ``repositories {}`` block in the ``build.gradle`` file. You will have to add this yourself. Above the ``dependencies {}`` block, please add the following:
+
+.. code-block:: groovy
+
+   repositories {
+      mavenCentral()
+      ...
+   }
+
+``mavenCentral()`` can be replaced with whatever repository the library you want to import is using. Now you have to add the dependency on the library itself. This is done by adding the necessary line to your ``dependencies {}`` block. The below example showcases adding Apache Commons to your gradle project.
```
Comment (Starlight220) [inline source/docs/software/advanced-gradlerio/external-libraries.rst:24]:
```suggestion
``mavenCentral()`` can be replaced with whatever repository the library you want to import is using. Now you have to add the dependency on the library itself. This is done by adding the necessary line to your ``dependencies {}`` block. The below example showcases adding Apache Commons to your Gradle project.
```
Same as above.

**Review by ThadHouse — APPROVED** [review]
